File: aware.blog.web.application/ClientApp/src/contract/dto/BlogPost.ts
import { Image } from './Image';
import { DateTime, int, NullableString } from '../../util/Types';
import { Entity } from './base/Entity';
import { User } from './User';
import { BlogPostTag } from './BlogPostTag';

export interface BlogPost extends Entity<int> {
    imageId: int;
    authorId: int;
    uri: string;
    title: string;
    description: NullableString;
    summary: NullableString;
    contentMarkdown: string;

    image: Image;
    author: User;
    tags: BlogPostTag[];
}

File: aware.blog.web.application/Domain/Entities/BlogPost.cs
using System.Collections.Generic;

namespace Aware.Blog.Domain
{
    public class BlogPost : Entity<int>
    {
        public int ImageId { get; set; }
        public int AuthorId { get; set; }
        public string Uri { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Summary { get; set; }
        public string ContentMarkdown { get; set; }

        public Image Image { get; set; }
        public User Author { get; set; }
        public IList<BlogPostTag> Tags { get; set; }

        public BlogPost()
        {
            Tags = new List<BlogPostTag>();
        }
    }
}